### DISCUSSION TITLE
PnP search box not showing results due to sorting

### DISCUSSION THREAD

**Discussion by Neenz13** [microsoft-search/pnp-modern-search#2834] [Q&A]
Hi there

just wanted to check if anyone knows a way to sort out search box results with sorting enabled. At the moment i have 2 search results that connect to the search box, data source is a sharepoint document library. Both these search results are sorted by Department first and then alphabetically sorted by file. This is so users don't get confused of which files belong to which department. I also have a filter in place so users can filter by department but searching also doesn't work correctly when enabled, due to the sorting being enabled. If I am searching for HR4.4 it will bring anything before this and put HR4.4 either in the 2nd page or at the bottom of the page (depending on how many files there are). Its like the sorting overrides the query template. Any suggestions would be great :)

![image](https://user-images.githubusercontent.com/126016838/224195535-4f7f5b2b-bf40-4207-b78b-81546ab53800.png)

![image](https://user-images.githubusercontent.com/126016838/224195612-95d87651-474b-4729-9b5b-4b22e5e19b1c.png)

![image](https://user-images.githubusercontent.com/126016838/224195771-a74cfeb9-894c-4e99-8ff8-d01863861913.png)

**Comment by wobba** [comment]
Can you try to get sorting working in e.g. the SharePoint Search Query Tool? Tried myself and seems to work fine, but sorting from the index seems to be case sensitive on `Filename`. What is not working with your current setup? Have you tried to sort on just one property?

And is it the search box or result web part not showing? What is your exact setup?

**Reply by Neenz13** [reply]
Hi @wobba, sorting is working perfectly fine. The search box does not work when the sorting settings are enabled. When sorting is not being used the search box searches perfectly fine, its just when i sort the documents in the search results by filename and department, search box does not like it and starts to order the files by the sorting settings.

Setup:
First search result web parts connect to a document library which has all documents but this web part is filtered to only show policies. This web part is sorted by Policy Area (refinablestring02) and then by Filename. This search result web part is connected to the search box using dynamic value
![image](https://user-images.githubusercontent.com/126016838/224570394-fc9094e0-b93d-46e2-ba82-28df2ad22f8b.png)


Second search result web part is also connected to the same document library as the first search result but filtered to only show the Associated documents (refinablestring08) which is also sorted, see image below. This search result web part is also connected to the search box using dynamic value
![image](https://user-images.githubusercontent.com/126016838/224570293-eab07a8a-f78b-4388-82cc-7f2e643aa0d7.png)

**Reply by wobba** [reply]
Can you better explains this statement: __"when i sort the documents in the search results by filename and department, search box does not like it and starts to order the files by the sorting settings"__?

My question is when is sorting not working per your setup? How would I repro sorting not working?

**Reply by Neenz13** [reply]
Hi wobba, i can close this one off and refer to #2834 as it is for the same thing. The issue is with the search box not the sorting
